Library/Tracking/KSRAnalytics.swift

```diff
@@ -533,24 +533,19 @@ public final class KSRAnalytics {
   }
 
   /// Configure Tracking Client's supporting user identity
-  private func identify(_ user: User?) {
-    guard let user = user else {
+  private func identify(oldUser: User?, newUser: User?) {
+    guard let newUser = newUser else {
       self.segmentClient?.reset()
       return
     }
 
-    let previousIdentityData = AppEnvironment.current.userDefaults.analyticsIdentityData
-
-    let newData = KSRAnalyticsIdentityData(user)
-
-    guard newData != previousIdentityData else { return }
+    guard newUser != oldUser else { return }
```

Comment:
Unfortunately due to our limited `Equatable` conformance in `User` (https://github.com/kickstarter/ios-oss/blob/master/KsApi/models/User.swift#L102), this will probably cause no `identify` calls to occur when notification options are updated.

I think we should just remove this de-duping altogether and optimize if it becomes an issue.

Reply:
Updated logic to compare the analytics data instead of the user data